### PR TITLE
create option to render diff buttons

### PR DESCRIPTION
**Pull request by enieber**
I migrate of react-native-swiper to this lib because performance in android, but I need adding buttons to change next or preview page, so this pr is for add option add leftButton and rightButton.

how to use:

```js
 <SwiperFlatList
        ref={ref => {
          this.flatListRef = ref
        }}
        horizontal
        pagingEnabled
        showPagination
        hasButton
        removeClippedSubviews
        scrollEnabled={false}
        data={this.state.onboardingSteps}
        index={this.state.step}
        paginationActiveColor={colors.primary}
        paginationDefaultColor={colors.gray}
        renderLeftButton={() => (
          <View style={style.containerButtonLeft}>
            <Button size="small" active onPress={this.prevStep.bind(this)}>
              <Icon name={'chevron-left'} size={30} color={colors.light} />
            </Button>
          </View>
        )}
        renderRightButton={() => (
          <View style={style.containerButtonRight}>
            <Button
              size="small"
              invalid={this.state.disableNext}
              active
              onPress={this.nextStep.bind(this)}
            >
              <Icon name={'chevron-right'} size={30} color={colors.light} />
            </Button>
          </View>
        )}
        renderItem={({ item, i }) => {
          return (
            <View>
               <Text>{item.name}</Text>
             </View>
          )
        }}
   />
```

### CODE REVIEW

**Comment by gusgard** [comment]
Thanks, @enieber, I will review it at the weekend.

File: src/components/Pagination/test.js [hunk 1]
```diff
@@ -1,23 +1,99 @@
 import React from 'react';
-import { render } from 'react-native-testing-library';
+import PropTypes from 'prop-types';
+import { StyleSheet, TouchableOpacity, View, ViewPropTypes } from 'react-native';
 
-import Pagination from './index';
+import { colors, vertical, horizontal } from '../../themes';
 
-describe('pagination', () => {
```
Comment (gusgard) [inline src/components/Pagination/test.js:6]: Could we keep the tests?

File: src/components/Pagination/index.js [hunk 2]
```diff
@@ -29,7 +33,31 @@ const Pagination = ({
   paginationActiveColor,
   paginationStyle,
   paginationStyleItem,
+  renderRightButton,
+  renderLeftButton,
+  hasButton,
 }) => {
+  if (hasButton) {
```
Comment (gusgard) [inline src/components/Pagination/index.js:40]: This should be done without the `hasButton`

```
{Array.from({ length: size }).map((_, index) => (
		{renderLeftButton?.()}
        <TouchableOpacity
          style={[
            styles.pagination,
            paginationStyleItem,
            paginationIndex === index
              ? { backgroundColor: paginationActiveColor }
              : { backgroundColor: paginationDefaultColor },
          ]}
          key={index}
          onPress={() => {
            scrollToIndex({ index });
          }}
        />
		{renderRightButton?.()}
      ))}
```

**Review by gusgard — CHANGES_REQUESTED** [review]
Thanks for this PR, could you check the comments?

File: src/components/SwiperFlatList/SwiperFlatList.js [hunk 3]
```diff
@@ -215,6 +218,9 @@ const SwiperFlatList = React.forwardRef(
       paginationDefaultColor,
       paginationStyle,
       paginationStyleItem,
+      renderRightButton,
```
Comment (gusgard) [inline src/components/SwiperFlatList/SwiperFlatList.js:221]: This should follow the name conventions `paginationRenderLeft` and `paginationRenderRight`

**Comment by enieber** [comment]
thanks @gusgard is code has many time ago, but I rewrite and update PR

File: src/components/Pagination/test.js (hunk 1 above)
Comment (enieber) [reply src/components/Pagination/test.js]: @gusgard the snapshot test is updated?

**Pull request closed without merging**